components/omega/src/ocn/Eos.cpp

```diff
@@ -140,6 +144,72 @@ void Eos::computeSpecVol(const Array2DReal &ConservTemp,
    }
 }
 
+/// Compute specific volume for all layer interfaces(no displacement)
+void Eos::computeSpecVolInterface(const Array2DReal &ConservTemp,
+                                  const Array2DReal &AbsSalinity,
+                                  const Array2DReal &PressureInterface) {
+   OMEGA_SCOPE(LocSpecVolInterface,
+               SpecVolInterface); /// Create a local view for computation
+   OMEGA_SCOPE(LocComputeSpecVolLinear,
+               ComputeSpecVolLinear); /// Local view for linear EOS computation
+   OMEGA_SCOPE(LocComputeSpecVolTeos10,
+               ComputeSpecVolTeos10); /// Local view for TEOS-10 computation
+   deepCopy(LocSpecVolInterface,
+            0); /// Initialize local specific volume to zero
+
+   I4 KDisp = 0; /// No displacement in this case
+
+   Array2DReal ConservTempInterface("ConservTempInterface", NCellsAll,
+                                    NVertLayers + 1);
+   Array2DReal AbsSalinityInterface("AbsSalinityInterface", NCellsAll,
+                                    NVertLayers + 1);
+
+   // Compute interface values by averaging adjacent layer values
+   parallelFor(
+       "compute-interface-values", {NCellsAll, NVertLayers + 1},
+       KOKKOS_LAMBDA(I4 ICell, I4 KInterface) {
+          if (KInterface == 0) {
```

Comment:
Shouldn't this be the equivalent of `minLevelCell` in MPAS-Ocean, rather than 0?

Reply:
Yes, I'll take care of swapping out for hierachical parallelism.